TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1702,16 +1704,31 @@ def _linemerge(geom):
             if not line.is_empty:
                 lines.append(line)
 
-        return linemerge(lines) if lines else MultiLineString([])
+        result_geom = linemerge(lines) if lines else None
 
     elif geom_type == 'LineString':
-        return geom
+        result_geom = geom
 
     elif geom_type == 'MultiLineString':
-        return linemerge(geom)
+        result_geom = linemerge(geom)
 
     else:
-        return MultiLineString([])
+        result_geom = None
+
+    if result_geom is not None:
+        # simplify with very small tolerance to remove duplicate points.
+        # almost duplicate or nearly colinear points can occur due to
+        # numerical round-off or precision in the intersection algorithm, and
+        # this should help get rid of those. see also:
+        # http://lists.gispython.org/pipermail/community/2014-January/003236.html
+        #
+        # the tolerance here is hard-coded to a fraction of the coordinate
+        # magnitude. there isn't a perfect way to figure out what this tolerance
+        # should be, so this may require some tweaking.
+        epsilon = max(map(abs, result_geom.bounds)) * float_info.epsilon * 1000
+        result_geom = result_geom.simplify(epsilon, True)
```

Comment:
We simplify until zoom 16, so the later simplification step wouldn't (and didn't) fix this at zoom 16. It seemed like this was worth having as the simplification step seems like it ought to be conceptually separate from the geometry merging, and that either should be independently able to produce valid results.

The simplification is about as un-aggressive as I could make it - on a mercator coordinate, it shouldn't ever be bigger than `4.45e-6`, and it's possible we could dial down that fixed factor of `1000` to something smaller. We'd just have to be careful to not re-introduce the duplicate points.

Of course, at this stage they're not _actually_ duplicates - they differed by `1.0e-9` - but that's close enough when we come to encode in MVT or GeoJSON. (But apparently not TopoJSON!)